account_payment_netting/models/account_payment.py

```diff
@@ -0,0 +1,101 @@
+        if ap and ar:  # Both AP and AR -> Netting
+            rec.update({'netting': True,
+                        'multi': False,  # With netting, allow edit amount
+                        'communication': ', '.join(invoices.mapped('number')),
+                        })
+        return rec
+
+    def _compute_journal_domain_and_types(self):
+        if not self.netting:
+            return super()._compute_journal_domain_and_types()
+        # For case netting, it is possible to have net amount = 0.0
+        # without forcing new journal type and payment diff handling
+        domain = []
+        if self.payment_type == 'inbound':
+            domain.append(('at_least_one_inbound', '=', True))
+        else:
+            domain.append(('at_least_one_outbound', '=', True))
+        return {'domain': domain, 'journal_types': set(['bank', 'cash'])}
+
+
+class AccountRegisterPayments(models.TransientModel):
+    _inherit = 'account.register.payments'
+
+    @api.multi
+    def get_payments_vals(self):
+        """ When doing netting, combine all invoices """
+        if self.netting:
+            return [self._prepare_payment_vals(self.invoice_ids)]
+        return super().get_payments_vals()
+
+    @api.multi
+    def _prepare_payment_vals(self, invoices):
+        """ When doing netting, partner_type follow payment type """
+        values = super()._prepare_payment_vals(invoices)
+        if self.netting:
+            values['netting'] = self.netting
+            values['communication'] = self.communication
+            if self.payment_type == 'inbound':
+                values['partner_type'] = 'customer'
+            elif self.payment_type == 'outbound':
+                values['partner_type'] = 'supplier'
+        return values
+
+    @api.multi
+    def create_payments(self):
+        if self.netting:
+            self._validate_invoice_netting(self.invoice_ids)
+        return super().create_payments()
+
+    @api.model
+    def _validate_invoice_netting(self, invoices):
+        """ Ensure valid selection of invoice for netting process """
+        # All invoice must be of the same partner
+        if len(invoices.mapped('partner_id')) > 1:
+            raise UserError(_('All invoices must belong to same partner'))
+        # All invoice must have residual
+        paid_invoices = invoices.filtered(lambda l: not l.residual)
+        if paid_invoices:
+            raise UserError(_('Some selected invoice are already paid: %s') %
+                            paid_invoices.mapped('number'))
+
+
+class AccountPayments(models.Model):
+    _inherit = 'account.payment'
+
+    @api.one
+    @api.depends('invoice_ids', 'payment_type', 'partner_type', 'partner_id')
+    def _compute_destination_account_id(self):
+        super()._compute_destination_account_id()
+        if self.netting:
+            if self.partner_type == 'customer':
+                self.destination_account_id = \
```

Comment:
(I'm not really aware of all accounting stuffs) Is it sufficent for refunds cases ?

Reply:
Yes, it should and I also make new menus specially when one want to pay net amount of invoice - refund

Reply:
@rousseldenis I found that the Oodo standard intentionally do not allow invoice and refund together
https://github.com/odoo/odoo/blob/12.0/addons/account/models/account_payment.py#L90
If I just remove this part of logic (and just create the menu that show invoice/payment together in list view), it will just work.

My opinion is, this should be a separated addon, i..e, account_payment_on_invoice_and_refund (still not a good name :p).

And the account_payment_netting is already completed by itself.

WDYT?

Reply:
I think the discussion here is related to this question https://github.com/OCA/account-invoicing/issues/551

Reply:
https://github.com/OCA/account-invoicing/pull/555